edk2toolext/edk2_invocable.py

```diff
@@ -151,12 +152,16 @@ def ParseCommandLineOptions(self):
         env = shell_environment.GetBuildVars()
         BuildConfig = os.path.abspath(args.build_config)
 
+        comment_re = re.compile(r"#.+$")
+
         if os.path.isfile(BuildConfig):
             with open(BuildConfig) as file:
                 for line in file:
                     stripped_line = line.strip()
-                    if not stripped_line.startswith("#"):
-                        unknown_args.append(line)
+                    stripped_line = comment_re.sub("", stripped_line)
```

Comment:
I am not a regex power user but the python partition function could work and would require no additional imports and fewer lines.  

Basically 
```python
stripped_line = stripped_line.partition("#")[0]
```

https://docs.python.org/3/library/stdtypes.html?highlight=partition#str.partition

Reply:
Good point